File: apps/csharp/FizzBuzz/FizzBuzz.cs
namespace FizzBuzz;

using System.Collections.Generic;
using System.IO;
using System.Linq;

public static class FizzBuzzRunner
{
    private const int FizzNumber = 3;
    private const int BuzzNumber = 5;

    public static string Generate(int number)
    {
        if (IsFizzBuzz(number))
        {
            return "FizzBuzz";
        }

        if (IsFizz(number))
        {
            return "Fizz";
        }

        if (IsBuzz(number))
        {
            return "Buzz";
        }

        return number.ToString();
    }

    public static List<string> GenerateList(int count)
    {
        return Enumerable.Range(1, count)
            .Select(Generate)
            .ToList();
    }

    public static void PrintFizzBuzz(TextWriter writer)
    {
        foreach (var item in GenerateList(100))
        {
            writer.WriteLine(item);
        }
    }

    private static bool IsFizz(int number) => number % FizzNumber == 0;

    private static bool IsBuzz(int number) => number % BuzzNumber == 0;

    private static bool IsFizzBuzz(int number) => IsFizz(number) && IsBuzz(number);
}

File: apps/kotlin/src/main/kotlin/fizzbuzz/FizzBuzz.kt
package fizzbuzz

/**
 * FizzBuzz を解くオブジェクト。
 */
object FizzBuzz {
    /**
     * 数 [n] を FizzBuzz の規則に従って文字列に変換する。
     */
    fun convert(n: Int): String = when {
        n % 15 == 0 -> "FizzBuzz"
        n % 3 == 0 -> "Fizz"
        n % 5 == 0 -> "Buzz"
        else -> n.toString()
    }

    /**
     * 1 から [n] までの FizzBuzz の結果をリストとして返す。
     */
    fun generateList(n: Int): List<String> = (1..n).map(::convert)
}
